src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -408,7 +408,7 @@ protected virtual void OutputVirtualSlots(NodeFactory factory, ref ObjectDataBui
             if (declType.HasGenericDictionarySlot())
             {
                 // Note: Canonical type instantiations always have a generic dictionary vtable slot, but it's empty
-                if (declType.IsCanonicalSubtype(CanonicalFormKind.Any))
+                if (declType.IsInterface || declType.IsCanonicalSubtype(CanonicalFormKind.Any))
```

Comment:
I would assume this actually being used if there's a static method on a generic interface. I wouldn't be surprised if we have no test coverage for that and it simply AV's (because the dictionary is null) in Project N.